src/devices/sonoff.ts

```diff
@@ -132,6 +133,182 @@ const sonoffExtend = {
             },
         }];
 
+        return {
+            exposes: [exposes],
+            fromZigbee,
+            toZigbee,
+            isModernExtend: true,
+        };
+    },
+    cyclicTimedIrrigation: (): ModernExtend => {
+        const exposes = e.composite('timedIrrigation', 'cyclic_timed_irrigation', ea.ALL)
+            .withDescription('Smart water valve cycle timing irrigation')
+            .withFeature(e.numeric('currentCount', ea.STATE).withDescription('Number of times it has been executed').withUnit('times'))
```

Comment:
```suggestion
            .withFeature(e.numeric('current_count', ea.STATE).withDescription('Number of times it has been executed').withUnit('times'))
```

Make all the features also `snake_case`